pybamm/expression_tree/concatenations.py

```diff
@@ -48,12 +48,13 @@ def get_children_domains(self, children):
         return domain
 
 
-class NumpyModelConcatenation(pybamm.Symbol):
-    """A node in the expression tree representing a concatenation of equations.
+class NumpyConcatenation(pybamm.Symbol):
+    """A node in the expression tree representing a concatenation of equations, when we
+    *don't* care about domains. The class :class:`pybamm.DomainConcatenation`, which
```

Comment:
For me it isn't clear exactly what we mean when we say *don't* care about domains. I'm probably missing some of the other functionality here but I can't think of when we wouldn't pass variables, or parameters without domains into a concatenation

Reply:
Or is this only used to create one big long rhs vector?

Reply:
Reading through more, NumpyConcatenation seems like quite a different thing to the other concatenations. Maybe we should name it differently to avoid confusion?